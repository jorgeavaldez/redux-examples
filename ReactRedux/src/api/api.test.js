import { getSubredditPosts } from './';

describe('The Reddit API', async () => {
  const subreddit = 'awww';

  it('should return a list of posts', async () => {
    const posts = await getSubredditPosts(subreddit);

    expect(posts);
    expect(posts.length);
    expect(posts.length > 0);
  });

  it('posts should contain the appropriate data', async () => {
    const posts = await getSubredditPosts(subreddit);

    expect(posts);
    expect(posts.length);
    expect(posts.length > 0);

    // gets the properties for the first post in the list
    // we already checked that there's more than 0 posts in the list
    const postProperties = Object.keys(posts[0].data);

    // start at 1 because we already checked the first post
    for (let i = 1; i < posts.length; i++) {
      const currentPostProperties = Object.keys(posts[i].data);

      // Make sure that the properties in the current post match
      // up with the first post
      expect(postProperties).toEqual(currentPostProperties);
    }

    expect(postProperties).toMatchSnapshot();
  });
});