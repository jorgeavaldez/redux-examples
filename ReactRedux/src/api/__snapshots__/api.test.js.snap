// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Reddit API posts should contain the appropriate data 1`] = `
Array [
  "domain",
  "approved_at_utc",
  "banned_by",
  "media_embed",
  "thumbnail_width",
  "subreddit",
  "selftext_html",
  "selftext",
  "likes",
  "suggested_sort",
  "user_reports",
  "secure_media",
  "is_reddit_media_domain",
  "link_flair_text",
  "id",
  "banned_at_utc",
  "view_count",
  "archived",
  "clicked",
  "report_reasons",
  "title",
  "num_crossposts",
  "saved",
  "mod_reports",
  "can_mod_post",
  "is_crosspostable",
  "pinned",
  "score",
  "approved_by",
  "over_18",
  "hidden",
  "preview",
  "thumbnail",
  "subreddit_id",
  "edited",
  "link_flair_css_class",
  "author_flair_css_class",
  "contest_mode",
  "gilded",
  "downs",
  "brand_safe",
  "secure_media_embed",
  "removal_reason",
  "post_hint",
  "author_flair_text",
  "stickied",
  "can_gild",
  "thumbnail_height",
  "parent_whitelist_status",
  "name",
  "spoiler",
  "permalink",
  "subreddit_type",
  "locked",
  "hide_score",
  "created",
  "url",
  "whitelist_status",
  "quarantine",
  "author",
  "created_utc",
  "subreddit_name_prefixed",
  "ups",
  "media",
  "num_comments",
  "is_self",
  "visited",
  "num_reports",
  "is_video",
  "distinguished",
]
`;
